### DISCUSSION TITLE
Reactive SQL Client, with SQL Server query return invalid negative number

### DISCUSSION THREAD

**Discussion by DevEnol** [quarkusio/quarkus#30628] [Q&A] [answered]
Hello everyone,

I am trying to use reactive SQL with Microsoft SQL Server 2019, sometimes the query result return invalid negative number for table with column of type  numeric(19,7).

Here are the steps to reproduce:
1. Create table

    CREATE TABLE master.dbo.test_bug (
    p_code varchar(100) NOT NULL,
    avg_buy numeric(19,7) NOT NULL,
    avg_sell numeric(19,7) NOT NULL,
    CONSTRAINT test_bug_PK PRIMARY KEY (p_code)
    )

2. Insert data

    INSERT INTO master.dbo.test_bug (p_code, avg_buy, avg_sell) VALUES ('ABCD', 325.0773780, 256.0000000);

    INSERT INTO master.dbo.test_bug (p_code, avg_buy, avg_sell) VALUES('WXYZ', 275.4000000, 286.0000000);

4. Perform query using Quarkus reactive SQL:

    @Inject
    MSSQLPool mssqlClient;

    public Multi<JsonObject> getBugList() {

        return mssqlClient.preparedQuery("SELECT * FROM test_bug").execute().onItem()

            .transformToMulti(set -> Multi.createFrom().iterable(set)).map(row ->
                JsonObject json = new JsonObject();
                json.put("pCode", row.getString(0));
                json.put("avgBuy", row.getBigDecimal(1));
                json.put("avgSell", row.getBigDecimal(2));
                return json;
        });
    }

The result returns invalid negative number:

    {
      "map": {
        "pCode": "ABCD",
        "avgBuy": -104.4193516,
        "avgSell": -173.4967296
      },
      "empty": false
    },
    {
      "map": {
        "pCode": "WXYZ",
        "avgBuy": -154.0967296,
        "avgSell": -143.4967296
      },
      "empty": false
    }

Not sure if my code is at fault here (Quarkus  version: 2.15.3)
Maybe this is  a bug in reactive SQL extension or SQL Server.

[a comment by an automated account is not shown]

**Comment by DevEnol** [comment, marked as answer]
My workaround to fix this, is to convert the column in the select statement: 

SELECT p_code, CAST(avg_buy AS VARCHAR) buy_avg, CAST(avg_sell AS VARCHAR) sell_avg FROM test_bug

**Reply by tsegismont** [reply]
@DevEnol this is due to https://github.com/eclipse-vertx/vertx-sql-client/issues/1272, I think.

It will be fixed in Vert.x version 4.3.8
